x/wasm-storage/keeper/msg_server.go

```diff
@@ -76,14 +82,15 @@ func (k msgServer) StoreOverlayWasm(goCtx context.Context, msg *types.MsgStoreOv
 }
 
 // unzipWasm unzips a gzipped Wasm into
-func unzipWasm(wasm []byte) []byte {
+func unzipWasm(wasm []byte) ([]byte, error) {
 	var unzipped []byte
 	var err error
-	if ioutils.IsGzip(wasm) {
-		unzipped, err = ioutils.Uncompress(wasm, types.MaxWasmSize)
-		if err != nil {
-			panic(err)
-		}
+	if !ioutils.IsGzip(wasm) {
+		return nil, fmt.Errorf("wasm is not gzip compressed")
```

Comment:
Change this so it returns an error when the wasm was not compressed, to fix the behavior "non gzip wasm files would be hash as nil"

Reply:
nice catch - tyty

Reply:
nice catch.. tyty